swagger_spec_validator/validator12.py

```diff
@@ -159,59 +170,67 @@ def validate_data_type(obj, model_ids, allow_arrays=True, allow_voids=False,
     raise SwaggerValidationError('no "$ref" or "type" present')
 
 
-def validate_model(model, model_name, model_ids):
+def validate_model(
+    model: dict[str, Any], model_name: str, model_ids: list[str]
+) -> None:
     """Validate a Model Object (§5.2.7)."""
     # TODO Validate 'sub-types' and 'discriminator' fields
-    for required in model.get('required', []):
-        if required not in model['properties']:
+    for required in model.get("required", []):
+        if required not in model["properties"]:
             raise SwaggerValidationError(
-                'Model "%s": required property "%s" not found' %
-                (model_name, required))
-
-    if model_name != model['id']:
-        error = 'model name: {} does not match model id: {}'.format(model_name, model['id'])
+                'Model "{}": required property "{}" not found'.format(
+                    model_name, required
+                )
+            )
+
+    if model_name != model["id"]:
+        error = "model name: {} does not match model id: {}".format(
+            model_name, model["id"]
+        )
         raise SwaggerValidationError(error)
 
-    for prop_name, prop in six.iteritems(model.get('properties', {})):
+    for prop_name, prop in model.get("properties", {}).items():
         try:
             validate_data_type(prop, model_ids, allow_refs=True)
         except SwaggerValidationError as e:
             # Add more context to the exception and re-raise
             raise SwaggerValidationError(
-                'Model "{}", property "{}": {}'.format(model_name, prop_name, str(e)))
+                'Model "{}", property "{}": {}'.format(model_name, prop_name, str(e))
+            )
 
 
-def validate_parameter(parameter, model_ids):
+def validate_parameter(parameter: dict[str, Any], model_ids: list[str]) -> None:
```

Comment:
very minor nit: you use `dict` in some places and `Dict` in others, i sort of like the lowercase style